src/AutoRest.CSharp/Mgmt/Generation/ArmResourceExtensionsWriter.cs

```diff
@@ -107,7 +107,7 @@ protected override void WriteResourceCollectionGetMethod(Resource resource)
                 resourceCollection.Type,
                 $"An object representing collection of {resource.Type.Name.LastWordToPlural()} and their operations over a {resource.Type.Name}.",
                 GetParametersForCollectionEntry(resourceCollection));
-            using (WriteCommonMethod(signature, null, false))
+            using (_writer.WriteCommonMethod(signature, null, false, This.Accessibility == "public"))
```

Comment:
shall we extract a string enum for `Accessiblity`?

Reply:
Not sure I follow.  I didn't want to change how mgmt plane determines if a method is "public" today.

Reply:
I believe Mingzhe means if we could introduce a new enum with Flags for the `Accessibility` property like MethodSignatureModifiers is currently doing, so that we do not have to maintain it in strings.

Reply:
I think we can do that, but I would like to leave this refactoring to a future PR as this PR doesn't introduce this and its just the first step, a big clean up PR will follow